cdc/changefeed.go

```diff
@@ -706,42 +726,34 @@ func (c *changeFeed) handleDDL(ctx context.Context) error {
 		c.ddlJobHistory = c.ddlJobHistory[1:]
 		c.ddlExecutedTs = todoDDLJob.BinlogInfo.FinishedTS
 		c.ddlState = model.ChangeFeedSyncDML
+		c.ddlEventCache = nil
 		return nil
 	}
 
-	executed := false
 	if !c.cyclicEnabled || c.info.Config.Cyclic.SyncDDL {
 		failpoint.Inject("InjectChangefeedDDLError", func() {
 			failpoint.Return(cerror.ErrExecDDLFailed.GenWithStackByArgs())
 		})
 
 		ddlEvent.Query = binloginfo.AddSpecialComment(ddlEvent.Query)
 		log.Debug("DDL processed to make special features mysql-compatible", zap.String("query", ddlEvent.Query))
-		err = c.sink.EmitDDLEvent(ctx, ddlEvent)
+
+		c.ddlEventCache = ddlEvent
+		cdcCtx := cdcContext.NewContext(ctx, &cdcContext.GlobalVars{})
```

Comment:
It is better if we do not create the `cdcCtx` afresh each time.

Reply:
done